client/asset/btc/btc.go

```diff
@@ -5090,3 +5092,71 @@ func deserializeMsgTx(r io.Reader) (*wire.MsgTx, error) {
 func msgTxFromBytes(txB []byte) (*wire.MsgTx, error) {
 	return deserializeMsgTx(bytes.NewReader(txB))
 }
+
+func (btc *baseWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, confs, err := btc.rawWalletTx(txHash)
+	// redemption transaction found, return its confirms.
+	if err == nil {
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  uint64(confs),
+			Req:    reqiredRedeemConfirms,
+			CoinID: coinID,
+		}, nil
+	}
+
+	if !errors.Is(err, WalletTransactionNotFound) {
+		return nil, fmt.Errorf("problem searching for redemption transaction %s: %v", txHash, err)
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. We
+	// assume a contract past its locktime cannot make it here, so it must
+	// not be refunded. Check if the contract is still an unspent output.
+
+	pkScript, err := btc.scriptHashScript(redemption.Spends.Contract)
+	if err != nil {
+		return nil, fmt.Errorf("error creating contract script: %v", err)
+	}
+
+	txHash, vout, err := decodeCoinID(redemption.Spends.Coin.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	utxo, _, err := btc.node.getTxOut(txHash, vout, pkScript, time.Now().Add(-ContractSearchLimit))
+	if err != nil {
+		return nil, fmt.Errorf("error finding unspent contract %s: %w", redemption.Spends.Coin.ID(), err)
```

Comment:
Just a nit: I noticed that here you are using `%w` to wrap this error while in other places you use `%v`, 

I'm curious whether you have some convention for this ? It seems both styles are used all across `client/asset/btc/btc.go` (I thought `%w` is usually preferred, unless you explicitly want to "disable" unwrapping for some reason)

Reply:
Probably %w is always correct for errors. I am just in the habit of using %v. Will change the error %v's to %w's

Reply:
%w is almost always the "right thing to do" but we've had an annoying issue where we unwrap some errors recursively and only report the inner error ...  https://github.com/decred/dcrdex/issues/1488
One of the most annoying DEX (non)issues TBH.

Reply:
v to w for errors https://github.com/decred/dcrdex/compare/53bd2f1384b8ba7c09ac0ff40302d7c1085de111..10f3413dbd2f2fdc612d6129e891a5792f2bb893